lib/straight-server/orders_controller.rb

```diff
@@ -13,17 +15,26 @@ def initialize(env)
     end
 
     def create
-      
+
       unless @gateway
         StraightServer.logger.warn "Gateway not found"
         return [404, {}, "Gateway not found" ]
       end
 
+      unless @gateway.check_signature
+        ip = @env['HTTP_X_FORWARDED_FOR'].to_s
```

Comment:
Ok, maybe we should add that somewhere into the documentation. Also, probably a good idea to just add a whole README section about throttling if you haven't done that already.